app/jobs/get_usps_proofing_results_job.rb

```diff
@@ -149,29 +159,51 @@ def handle_bad_request_error(err, enrollment)
     else
       analytics(user: enrollment.user).idv_in_person_usps_proofing_results_job_exception(
         **enrollment_analytics_attributes(enrollment, complete: false),
-        **response_analytics_attributes(err.response),
-        reason: 'Request exception',
+        **response_analytics_attributes(response),
         exception_class: err.class.to_s,
         exception_message: err.message,
+        reason: 'Request exception',
+        response_status_code: err.response_status,
       )
       enrollment_outcomes[:enrollments_errored] += 1
     end
   end
 
-  def handle_standard_error(err, enrollment)
-    response_attributes = if err.respond_to?(:response)
-                            response_analytics_attributes(err.response)
-                          else
-                            response_analytics_attributes(nil)
-                          end
+  def handle_client_or_server_error(err, enrollment)
+    analytics(user: enrollment.user).idv_in_person_usps_proofing_results_job_exception(
+      **enrollment_analytics_attributes(enrollment, complete: false),
+      **response_analytics_attributes(err.response_body),
+      exception_class: err.class.to_s,
+      exception_message: err.message,
+      reason: 'Request exception',
+      response_status_code: err.response_status,
+    )
+    enrollment_outcomes[:enrollments_errored] += 1
+  end
+
+  def handle_faraday_error(err, enrollment)
+    response_status_code = err.response_status || err&.response&.status
```

Comment:
Does `err.response` respond to a `status` method? Based on what I can tell from Faraday internals, it would be a hash.

https://github.com/lostisland/faraday/blob/ceeaa7476f7f9c2ccf1631aa2bfdd970ca469974/lib/faraday/response/raise_error.rb#L40-L54

https://github.com/lostisland/faraday/blob/ceeaa7476f7f9c2ccf1631aa2bfdd970ca469974/lib/faraday/error.rb#L32

Should we leave out the fallback altogether? Or am I missing a case that it's intending to cover?

Reply:
I found one case where the fallback actually works: if we encounter a Faraday::ParsingError you have to dig into `err.response[:status]`. For some reason `err.response_status` is just `nil`

Reply:
And just to be totally clear: for most types of `Faraday::Errors` not caught elsewhere `err.response` is `nil` but in the case of `Faraday::ParsingError` `err.response` is a `Faraday::Response`. So that's why I'm using the safe-navigation operator in `err&.response&.status`

But I believe I actually only need one safe-navigation operator like `err.response&.status`

Reply:
Hm, this feels like an odd behavior of Faraday (maybe even a bug?) to be initializing error responses sometimes as a hash and other times as a `Faraday::Response`. I'd guess that also explains why `error.response_status` is `nil` for the parsing errors, since it's trying to access the status as if response were a hash, and it's not. If it is a bug, I'd worry if this could change under our feet in the future. Another worry is that if we can't really trust what `err.response` is, I don't have a whole lot of trust that `err.response.status` won't be trying to call a non-existent `status` method on a Hash object.

I wonder if a compromise could be to try to coerce whatever it is into a hash and access the status from there?

```rb
err.response.to_hash[:status]
```

Reply:
Hmm yeah that is weird. Of all of the cases we test for it's only `Faraday::ParsingError` which sets `err.response` to a `Faraday::Response` instance. Everywhere else is a hash

I think I'll backtrack and suggest we just remove the fallback entirely

Reply:
Removed in [36e2ff5](https://github.com/18F/identity-idp/pull/7212/commits/36e2ff5590d80cd91f29e34e93060fd33de809c7)